contrib/README.md

```diff
@@ -0,0 +1,84 @@
+
+- **deploy.sh**
+
+    This script is the main script to install the feature, it should be called by the task from the Tekton pipeline, and should contain all the step required to deploy the requried feature in the spoke cluster. Before start any step it should be sure that the feature is not already installed, this is done by calling the verify.sh script. If the stderr of the verify.sh script is `0`, the feature is already installed and the deploy.sh script should not be executed.
+
+- **manifests**
+
+    Within the folder `manifests` should the .yaml files to be applied to the spoke cluster. In the template example is just a manifest to create a namespace called `contrib-template`. In here can be created as must as needed manifests objects to be applied to the spoke cluster, called from the `deploy.sh` script.
+
+# 2. Create contrib feature 
+
+The first thing to do is to create a directory for the contrib features under the `contrib` folder, using as base the `contrib/template` example. The directory name should be the same as the feature name. In the example, the name of the directory can be `contrib/deploy-app`.
+
+```bash
+cd contrib
+cp -r template deploy-app
+```
+
+At this point all the scripts and manifests should be updated with the required instructions to deploy the feature in the spoke cluster.
+
+# 3. Create pipeline tasks
+
+Once the contrib feature is created, the next step is to create the Tekton tasks to call the deploy script. The tasks should be created in the `pipelines/resources/contrib/` folder.
+The current template example can be copied to the folder and renamed to `deploy-app`.
+
+```bash
+cp pipelines/resources/contrib/contrib-template.yaml pipelines/resources/contrib/deploy-app.yaml
+```
+
+After copy the task definition from the template, can be updated with the required instructions to call the previously created deploy.sh script and define the variables and/or config if required.
+
+Be aware to add to the `kustomization.yaml` the name of the task reference name.
+
+# 4. Add tasks to the spoke pipeline
+
+The last step is to add the task to the pipeline. The task should be added to the spoke pipelines file at `pipelines/resources/deploy-ztp-spokes.yaml`. The position of the call to the new task depend of the nature of the feature and the dependencies of the features from other steps in the pipeline.
+
+The snippet below should be added in the `deploy-ztp-spokes.yaml` updating the values that fit for the desired task run. It is important to be carefull with the value of `name.runAfter`, in here has to be defined the previous task of our run in the pipeline.
+
+```yaml
+- name: contrib-template
+    taskRef:
+      name: spoke-contrib-template
+    params:
+      - name: spokes-config
+        value: $(params.spokes-config)
+      - name: kubeconfig
+        value: $(params.kubeconfig)
+      - name: ztp-container-image
+        value: $(params.ztp-container-image)
+      - name: mock
```

Comment:
"mock" should be removed as well as you did in the script code. Just to make it easy for them reducing extra params never will be used by them